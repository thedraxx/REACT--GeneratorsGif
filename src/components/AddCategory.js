import React, { useState } from 'react'
import PropTypes from 'prop-types';

export const AddCategory = ( {setCategories} ) => {

    const [inputValue, setinputValue] = useState("Search a GIF!");

    const handeInputChange = (e) =>{
        setinputValue(e.target.value)

        console.log('handeInputChange llamado')
    }

    const handleSubmit = (e) => {
        
        e.preventDefault();
  
        if( inputValue.trim().length > 2){
            setCategories(categories => [ inputValue,...categories,]);
            setinputValue('');
        }
    }

    return (
        <form onSubmit = {handleSubmit}>
            <p>{inputValue}</p>
                <input
                    type="text" 
                    value={inputValue}
                    onChange={ handeInputChange}
                />

        </form>
    )
}

AddCategory.propTypes = {
    setCategories: PropTypes.func.isRequired
  };
