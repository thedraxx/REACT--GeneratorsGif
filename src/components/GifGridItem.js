import React from 'react'
import PropTypes from 'prop-types';

export const GifGridItem = ({tittle,url}) => {

    return (
        <div className = "card animate__animated animate__fadeIn">
           <img src = {url} alt={tittle} />
        </div>
    )
}

GifGridItem.propTypes = {
    tittle: PropTypes.string.isRequired,
    url: PropTypes.string.isRequired
}



