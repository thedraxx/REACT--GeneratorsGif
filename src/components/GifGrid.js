import React from 'react';
import PropTypes from 'prop-types';
import { useFetchGifs } from '../hooks/useFetchGifs';
import { GifGridItem } from './GifGridItem';
import '../index.css';

export const  GifGrid = ({category}) => {

    const {data:images, loading} = useFetchGifs(category);

    return (
        <>
        <div className='Contenedor-titulo'>
          
       
            {loading && <p className= "card animate__animated animate__flash">Loading</p> }
       
            <div className="card-grid">
                {
                    images.map(img => (
                        <GifGridItem 
                                key={img.id}
                                {...img}
                            />
                    ))
                }
            </div>
        </div>
        </>
    )
}

GifGrid.propTypes = {
    category: PropTypes.string.isRequired
}
