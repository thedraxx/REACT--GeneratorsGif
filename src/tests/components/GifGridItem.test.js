import React from 'react';
import { shallow } from 'enzyme';
import { GifGridItem } from '../../components/GifGridItem';

describe('Pruebas en <GifGridItem />', () => {

    const tittle = 'Un Titulo';
    const url = 'https://localhost/algo.jpg';
    const wrapper = shallow(<GifGridItem  tittle = {tittle} url = {url}/>);
    
    test('debe mostrar el test Correctamente', () => {
        
        expect (wrapper).toMatchSnapshot();
    })
    

    test('Debe de tener la imagen igual al url y alt de los props', () => {
        
        const img = wrapper.find('img');
        expect(img.prop('src')).toBe(url);
        expect(img.prop('alt')).toBe(tittle);
    });

    test('debe de tener un animate__fadein', () => {
        
        const div = wrapper.find('div');
        const valor = div.prop('className');
        let palabras = valor.split(' ');
        expect(palabras[2]).toBe('animate__fadeIn');
    })
    
    


    
})
