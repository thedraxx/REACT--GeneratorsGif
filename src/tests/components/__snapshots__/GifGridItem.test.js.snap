// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <GifGridItem /> debe mostrar el test Correctamente 1`] = `
<div
  className="card animate__animated animate__fadeIn"
>
  <img
    alt="Un Titulo"
    src="https://localhost/algo.jpg"
  />
</div>
`;
